iocage_lib/ioc_fstab.py

```diff
@@ -61,41 +63,155 @@ def __init__(self, uuid, action, source, destination, fstype, fsoptions,
         self.callback = callback
 
         if action != "list":
+            self.fstab = list(self.__read_fstab__())
+            self.dests = self.__validate_fstab__(self.fstab, 'all')
             self.__fstab_parse__()
 
     def __fstab_parse__(self):
         """
         Checks which action the user is asking for and calls the
         appropriate methods.
         """
+        actions = ['add', 'remove', 'edit', 'replace']
+
+        if self.action not in actions:
+            raise RuntimeError("Type of operation not specified!")
 
         if self.action == "add":
+            self.__validate_fstab__([self.mount])
             self.__fstab_add__()
-            self.__fstab_mount__()
+
+            try:
+                self.__fstab_mount__()
+            except RuntimeError:
+                iocage_lib.ioc_common.logit({
+                    'level': 'WARNING',
+                    'message': 'Mounting entry failed, check \'mount\''
+                },
+                    _callback=self.callback,
+                    silent=self.silent
+                )
         elif self.action == "remove":
             dest = self.__fstab_remove__()
-            self.__fstab_umount__(dest)
+
+            try:
+                self.__fstab_umount__(dest)
+            except RuntimeError:
+                iocage_lib.ioc_common.logit({
+                    'level': 'WARNING',
+                    'message': 'Unmounting entry failed, check \'mount\''
+                },
+                    _callback=self.callback,
+                    silent=self.silent
+                )
         elif self.action == "edit":
             self.__fstab_edit__()
         elif self.action == "replace":
+            self.__validate_fstab__([self.mount])
             self.__fstab_edit__(_string=True)
             self.__fstab_mount__()
-        else:
-            raise RuntimeError("Type of operation not specified!")
+
+    def __read_fstab__(self):
+        with open(f"{self.iocroot}/jails/{self.uuid}/fstab", "r") as f:
+            for line in f:
+                yield line.rstrip()
+
+    def __get_fstab_dests_(self):
+        dests = []
+
+        for line in self.fstab:
+            source, destination, fstype, options, \
+                dump, _pass = line.split()[0:6]
+            dests.append(destination)
+
+        return dests
+
+    def __validate_fstab__(self, fstab, mode='single'):
+        dests = []
+        verrors = []
+        jail_root = f'{self.iocroot}/jails/{self.uuid}/root'
+
+        for line in fstab:
+            source, destination, fstype, options, \
+                dump, _pass = line.split()[0:6]
```

Comment:
This will raise an exception if the fstab file is not configured properly. We would like to raise one obviously in that case but this will specifically raise valueerror if the values are less then 6. Would that be ideal ?